server/routes/incidentDetails/incidentDetailsPage.ts

```diff
@@ -0,0 +1,728 @@
+      ...stashedData.incidentDetails,
+    },
+    temporaryData: stashedData.temporaryData,
+  }
+}
+
+const getDraftIdFromString = (draftId: string): number => {
+  let draftIdValue: number = null
+  if (draftId) {
+    draftIdValue = parseInt(draftId, 10)
+  }
+  return draftIdValue
+}
+
+const radioSelectionCodeFromIncidentRole = (incidentRole: IncidentRole): string => {
+  if (!incidentRole) {
+    return null
+  }
+  let radioSelectionCode = null
+  switch (incidentRole) {
+    case IncidentRole.COMMITTED:
+      radioSelectionCode = 'committed'
+      break
+    case IncidentRole.ATTEMPTED:
+      radioSelectionCode = 'attempted'
+      break
+    case IncidentRole.INCITED:
+      radioSelectionCode = 'incited'
+      break
+    case IncidentRole.ASSISTED:
+    // Fall through
+    default:
+      radioSelectionCode = 'assisted'
+      break
+  }
+  return radioSelectionCode
+}
+
+const incidentRoleFromRadioSelection = (radioSelectionCode: string): IncidentRole => {
+  if (!radioSelectionCode) {
+    return null
+  }
+  if (radioSelectionCode === 'attempted') {
+    return IncidentRole.ATTEMPTED
+  }
+  if (radioSelectionCode === 'incited') {
+    return IncidentRole.INCITED
+  }
+  if (radioSelectionCode === 'assisted') {
+    return IncidentRole.ASSISTED
+  }
+  return IncidentRole.COMMITTED
+}
+
+const isReviewer = (originalPageReferrer?: string): boolean => {
+  return originalPageReferrer?.includes('review')
+}
+
+const getIncidentDate = (userProvidedValue?: SubmittedDateTime) => {
+  if (userProvidedValue) {
+    const {
+      date,
+      time: { hour, minute },
+    } = userProvidedValue
+    return { date, hour, minute }
+  }
+  return null
+}
+
+const getTaskListUrl = (prisonerNumber: string, draftId: number) => {
+  return `/place-the-prisoner-on-report/${prisonerNumber}/${draftId}`
+}
+
+const redirectToSearchForPersonPage = (res: Response, searchTerm: string) => {
+  return res.redirect(`/select-associated-prisoner?searchTerm=${searchTerm}`)
+}
+
+const redirectToDeletePersonPage = (res: Response, prisonerToDelete: string) => {
+  return res.redirect(`/delete-person?associatedPersonId=${prisonerToDelete}`)
+}
+
+const redirectToOffenceSelection = (res: Response, draftId: number, incidentRoleCode: IncidentRole) => {
+  return res.redirect(`/offence-code-selection/${draftId}/${radioSelectionCodeFromIncidentRole(incidentRoleCode)}`)
+}
+
+const redirectToCheckYourAnswers = (
+  res: Response,
+  prisonerNumber: string,
+  draftId: number,
+  isReviewerPage: boolean
+) => {
+  return res.redirect(`/check-your-answers/${prisonerNumber}/${draftId}/${isReviewerPage ? 'review' : 'report'}`)
+}
+
+const redirectToTaskList = (res: Response, prisonerNumber: string, draftId: number) => {
+  return res.redirect(getTaskListUrl(prisonerNumber, draftId))
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const setUpRedirectForEditError = (res: Response, prisonerNumber: string, error: any, draftId: number) => {
```

Comment:
Could error be typed as an Error? or is potentially something else?

Reply:
I couldn't see anything specific in the catch documentation. Suppose I could debug it?!